services/sources/dspace_service.py

```diff
@@ -51,7 +51,10 @@ def get_records(self, full_import=False, start_timestamp=None, offset: Optional[
 
                 try:
                     parsed_record = self.parse_record(record)
-                    mapped_records.append(parsed_record)
+                    if parsed_record.record:
+                        mapped_records.append(parsed_record)
+                    else:
```

Comment:
Sorry, missed this on the first go through. This else condition is not needed since it will always continue after the if statement is evaluated.

Reply:
I thought so as well until my CLACSO integration test kept getting an AssertionError for the len(records) due to the `parsed_record.record` being None and thus returning an mapped_record array. Adding this continue statement makes it so we only process the records that are fully mapped and check the conditional statement once the if conditional is fulfilled. The continue statement will start the next iteration of the for loop: `for record in oaidc_records.xpath('//oai_dc:dc', namespaces=self.OAI_NAMESPACES)`